javascript/tool/js/rich-map.js

```diff
@@ -9,13 +9,145 @@ var map = new mapboxgl.Map({
   preserveDrawingBuffer: true
 });
 
+// geoJSONPolygons = a geoJSON FeatureCollection where the Features have a 'geometry'
+// object of type 'Polygon'.
+// aggregateData = a json object resulting from a query for aggregate data from the 
+// project's Flask API.
+// reconcileGroupNames = a callback with two arguments: (a) an element within the array of 
+// objects returned from an API call. and (b) a geoJSON polygon feature. The callback
+// returns true if the zone name within (a) matches the zone name within (b). 
+
+function addDataToPolygons(geoJSONPolygons, aggregateData, zoneNamesMatch){
```

Comment:
Note for future refactoring - do we want to move this up to the housing-insights.js level? This would be useful on any map that needs polygons.